lock.py

```diff
@@ -0,0 +1,26 @@
+import os
+import threading
+
+
+def number_generator(thread_number):
+    global counter
+    while True:
+        lock.acquire()
+        if counter < 101:
+            if ((counter % 2) == (thread_number % 2)):
+                print('Thread ' + str(thread_number) + ': ' + str(counter))
```

Comment:
`print` concatenates arguments to one string, str is applied  implicitly:
`print('Thread', thread_number, ':', counter)` or
'print('Thread %d: %d ' % (thread_number, counter))` or
'print('Thread {}: {} '.format(thread_number, counter))`